File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Floor.java
package uk.ac.uea.mathsthing.plugins.functions;

import java.math.BigDecimal;

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin;
import uk.ac.uea.mathsthing.util.FormulaException;

public class Floor extends IFunctionPlugin {
	
	@Override
	public String getName()
	{
		return "floor";
	}

	@Override
	public BigDecimal function(BigDecimal input) throws FormulaException
	{
		return new BigDecimal(Math.floor(input.doubleValue()));
	}

}

File: MathsThing_Plugins/src/uk/ac/uea/mathsthing/plugins/functions/Floor.scala
package uk.ac.uea.mathsthing.plugins.functions

import uk.ac.uea.mathsthing.IPlugin.IFunctionPlugin
import java.math.BigDecimal
import uk.ac.uea.mathsthing.Functions

class Floor extends IFunctionPlugin {

  def getName : String = "floor"
    
  def function(input: BigDecimal) : BigDecimal = {
    val result:BigDecimal = new BigDecimal(Math.floor(input.doubleValue))
    result
  }
}